http-core/src/main/scala/org/apache/pekko/http/impl/engine/http2/HttpMessageRendering.scala

```diff
@@ -99,7 +99,7 @@ private[http2] sealed abstract class MessageRendering[R <: HttpMessage] extends
       r.attribute(AttributeKeys.trailer) match {
         case Some(trailer) if trailer.headers.nonEmpty =>
           OptionVal.Some(ParsedHeadersFrame(streamId, endStream = true, trailer.headers, None))
-        case None => OptionVal.None
+        case _ => OptionVal.None
```

Comment:
So the question here is if someone added a `AttributeKeys.trailer` but it contained no headers, should we produce a `ParsedHeadersFrame` here? Looking at `Http2SubStream`/`OutStreamImpl`, it does seem safe (it will produce an empty `DataFrame` with `endStream = true` if necessary). It seems unlikely that any downstream implementation would rely on receiving an empty frame of trailing headers.

All in all I think this is OK, though adding a `AttributeKeys.trailer` without any headers seems strange, the code that caused that might deserve a closer look.